api/v1alpha1/httproute_types.go

```diff
@@ -190,6 +200,75 @@ type HTTPRouteAction struct {
 	Extension *core.TypedLocalObjectReference `json:"extension"`
 }
 
+// TLSConfig defines the schema of a TLS configuration.
+type TLSConfig struct {
+	// Termination defines how to terminate TLS connections.
+	//
+	// If unspecified, TLS termination type "Edge" will be used.
+	//
+	// Support: core
+	//
+	// +optional
+	Termination TLSTerminationType `json:"termination,omitempty"`
+
+	// Certificate is a reference to a Kubernetes object containing
+	// the identity certificate, key and CA certificate used to terminate
+	// the TLS connection. Certificate is used with "Reencrypt" and "Edge"
+	// TLS termination types.
+	//
+	// If apiGroup and kind are empty, will default to a Kubernetes Secret
+	// resources.
+	//
+	// If unspecified, the certificate of the binding gateway will be used.
+	//
+	// Support: Core (Kubernetes Secrets)
+	// Support: Implementation-specific (For other resource types)
+	//
+	// +optional
+	Certificate core.TypedLocalObjectReference `json:"certificate,omitempty"`
```

Comment:
Same comment as: https://github.com/kubernetes-sigs/service-apis/pull/35#discussion_r373215812

an HTTPRoute could have multiple SNIs.

Reply:
There's also other cases for allowing multiple server certificates (e.g. serving different type types).

Reply:
> serving different type types

Not sure I understand. Can you elaborate?

Reply:
A single application, served over HTTPs that has a collection of server certificates with different key types. The client will negotiate the matching certificate.

For example, [envoy TLS context](https://www.envoyproxy.io/docs/envoy/v1.6.0/api-v2/api/v2/auth/cert.proto#auth-commontlscontext) lets you add ECDSA and RSA certificates, at least in theory, the docs are contradictory :)